packages/SwingSet/src/kernel/vat-loader/manager-subprocess-xsnap.js

```diff
@@ -238,11 +240,49 @@ export function makeXsSubprocessFactory({
      * @returns {Promise<SnapshotResult>}
      */
     async function makeSnapshot(snapPos, snapStore, restartWorker) {
+      async function saveSnapshot(vatID_, snapPos_, snapshotStream_) {
+        const saveResultP = snapStore.saveSnapshot(
+          vatID_,
+          snapPos_,
+          snapshotStream_,
+        );
+        saveResultP
+          .then(
+            ({
+              hash: snapshotID,
+              uncompressedSize,
+              dbSaveSeconds,
+              compressedSize,
+              compressSeconds,
+            }) => {
+              const uncompressedSizeDelta =
+                uncompressedSizeLoaded &&
+                uncompressedSize - uncompressedSizeLoaded;
+              kernelSlog.write({
+                type: 'heap-snapshot-save',
+                vatID: vatID_,
+                snapshotID,
+                uncompressedSize,
+                dbSaveSeconds,
+                compressedSize,
+                compressSeconds,
+                endPosition: snapPos_,
+                restartWorker,
+                uncompressedSizeDelta,
+              });
+            },
+          )
+          .catch(err => {
+            throw err;
+          });
+
+        return saveResultP;
+      }
       const snapshotDescription = `${vatID}-${snapPos}`;
       const snapshotStream = worker.makeSnapshotStream(snapshotDescription);
```

Comment:
Simplification:
```suggestion
      const snapshotDescription = `${vatID}-${snapPos}`;
      const snapshotStream = worker.makeSnapshotStream(snapshotDescription);
      const saveSnapshot = async () => {
        const results = await snapStore.saveSnapshot(
          vatID,
          snapPos,
          snapshotStream,
        );
        const { hash: snapshotID, ...metrics } = results;
        const uncompressedSizeDelta =
          uncompressedSizeLoaded &&
          metrics.uncompressedSize - uncompressedSizeLoaded;
        kernelSlog.write({
          type: 'heap-snapshot-save',
          vatID,
          snapshotID,
          endPosition: snapPos,
          ...metrics,
          restartWorker,
        });
        return results;
      };
```
This also has the benefit of adding `archiveWriteSeconds` to the slog entry.